src/views/domain-page/domain-page-metadata/domain-page-metadata.tsx

```diff
@@ -1,7 +1,38 @@
+'use client';
 import React from 'react';
 
+import { useSuspenseQuery } from '@tanstack/react-query';
+
+import ListTable from '@/components/list-table/list-table';
+import PageSection from '@/components/page-section/page-section';
+import request from '@/utils/request';
+
+import domainPageMetadataTableConfig from '../config/domain-page-metadata-table.config';
 import { type DomainPageTabContentProps } from '../domain-page-content/domain-page-content.types';
+import { type DomainInfo } from '../domain-page.types';
+
+import { styled } from './domain-page-metadata.styles';
 
 export default function DomainPageMetadata(props: DomainPageTabContentProps) {
-  return <div>Metadata tab</div>;
+  const { data: domainInfo } = useSuspenseQuery<DomainInfo>({
+    queryKey: ['describeDomain', props],
+    queryFn: () =>
+      request(`/api/domains/${props.domain}/${props.cluster}`).then((res) => {
+        if (!res.ok) {
```

Comment:
hmm should we at least console.log real error? The current approach of showing only "Failed to fetch" hides all insights from the user which is hard to understand.

Reply:
I'm working on a more detailed error framework to show more info to the user.